Adding guidance for replaceable values (#30)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/general-formatting.adoc

When the format _day Month year_ causes a presentation or clarity issue, use instead: _Month day, year_ (for example, October 3, 2019).

[[user-replaced-values]]
== User-replaced values

> A _user-replaced value_, or _replaceable value_, is a value that the user must replace with a value that is relevant for their situation. User-replaced values are often found in places such as code blocks, file paths, and commands.

Reviewer: @bergerhoffer 
1. Generally only use one term, unless there's a good reason to use more than one. Since you're defining the terms here, is there any reason to include both terms _user-replaced value_ and _replaceable value_?
2. Please add a short example. Suggestion:
```suggestion
A _user-replaced value_  is a value that the user must replace. For example, in the following command `_filename_` and `_destination_` are user-replaced values.

`cp _filename_ _destination_`

User-replaced values are often found in places such as code blocks, file paths, and commands.
```

Author: @stoobie I updated to remove the second term mention of "replaceable value". 

The examples are further down after we explain the guidelines. Are those not sufficient?

> A _user-replaced value_, or _replaceable value_, is a value that must be replaced by the user with a value that is relevant for their situation. User-replaced values are often found in places such as code blocks, file paths, and commands.

Reviewer 2: ```suggestion
A _user-replaced value_, or _replaceable value_, is a value that the user must replace with a value that is relevant for their situation. User-replaced values are often found in places such as code blocks, file paths, and commands.
```
Suggestion to make passive construction, active.

> User-replaced values should use a descriptive names, and should follow this general format: _<value_name>_.

Reviewer 2: ```suggestion
Use descriptive names for user-replaced values and follow this general format: _<value_name>_.
```

Suggestion to make the sentence more user-action oriented and prescriptive and replace the 'should' so it's more direct.

> Ensure that user-replaced values have following characteristics:

Reviewer 2: ```suggestion
Ensure that user-replaced values have the following characteristics:
```

One more tiny suggestion to add 'the' back in.

Author: Whoops, good catch sorry!

> Replaceable values should have the following characteristics:

Reviewer 2: To replace the 'should' to be more direct per IBM, you could change this to something like one of these options:

- Use the following format for user-replaced values:
- Replaceable values must have the following characteristics:
- Ensure that user-replaced values have following characteristics:

It might also be nicer to keep referring to them as 'user-replaced values' for consistency throughout the body of the entry?

* Surrounded by angle brackets (`< >`)
* Separated by underscores (`_`) for multi-word values
* Lowercase, unless the rest of the related text is uppercase or another capitalization scheme.
* Italicized

> * If the replaceable value is referencing a value in code or in a command that would normally be monospace, then the reference to the replaceable value should also be monospace.

Reviewer 2: ```suggestion
* If the replaceable value is referencing a value in code or in a command that is normally monospace, use monospace for the replaceable value also.
```

Suggestion to be more direct and prescriptive.

.Example AsciiDoc: User-replaced value in a paragraph

----
Create an Ansible inventory file that is named `/_<path>_/inventory/hosts`.
----

This renders as follows:

====
Create an Ansible inventory file that is named `/_<path>_/inventory/hosts`.
====

> To italicize a user-replaced value in a code block, you must add `subs="+quotes"` to the attribute list of the code block.

Reviewer: @bergerhoffer I'm not an expert on the code block macros, but are you positive this is a requirement? Here's an example from our docs:
```
[options="nowrap" subs="normal"]
----
Host fully qualified DNS name of this server [_autodetected host name_]:
----
```
And I _think_ that when you have additional underscores in the variable, you should (must?) use double-underscores for italicizing, e.g.: __multi_word_variable__

If I'm wrong, I'd love to learn more.

Author: @stoobie In order for the italics to render, you need to add something to the attribute list to tell it to do so. `quotes` is the attribute that performs text styling substitutions. It looks like your example uses the `normal` attribute, which looks like it adds text styling and all other substitutions except callouts [1]. So yes the `normal` one seems to work fine too, but I'm not sure if we'd want to use that one as the suggestion, as it performs additional substitutions that might not be wanted. Let me know if you do think we should mention it as an alternative though.

[1] https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#attributes-and-substitutions

As for the double underscores, you definitely don't have to use them if the word has underscores, but you are right that there are some situations where the double underscores might be necessary. I can't remember exactly what those are - I think it might be when there is some additional formatting. However, I don't see any harm in using the double underscores in our example. I'll update.

Reviewer: @bergerhoffer It sounds like you have a better grasp on the code block macros than I do. However, from what you wrote, it's not accurate to say that you must add `subs="+quotes"` if there are other alternatives. I think it would be nice to say that there are alternatives, and include a link to the quick reference you included here.

Author: @stoobie That page doesn't have a great anchor for that attribute section, so I don't necessarily want to link to that. But I've updated to include both quotes and normal as examples of attributes that apply text formatting. Let me know what you think!

> To italicize a replaceable value in a code block, you must add `subs="+quotes"` to the code block's attribute list.

Reviewer 2: ```suggestion
To italicize a replaceable value in a code block, you must add `subs="+quotes"` to the attribute list of the code block.
```

Suggestion to be more fully in line with IBM style.
